algorithms/kernel/dtrees/forest/regression/oneapi/df_regression_train_dense_default_oneapi_impl.i

```diff
@@ -0,0 +1,1245 @@
+        DAAL_CHECK_STATUS_VAR(status);
+
+        context.run(range, kernel, args, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    return status;
+}
+
+template <typename algorithmFPType, decision_forest::regression::training::Method method>
+services::Status RegressionTrainBatchKernelOneAPI<algorithmFPType, method>::partitionCopy(UniversalBuffer & treeOrderBuf, UniversalBuffer & treeOrder,
+                                                                                          size_t iStart, size_t nRows)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.partitionCopy);
+
+    services::Status status;
+
+    auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
+
+    auto & kernel = kernelPartitionCopy;
+
+    {
+        KernelArguments args(3);
+        args.set(0, treeOrderBuf, AccessModeIds::read);
+        args.set(1, treeOrder, AccessModeIds::write);
+        args.set(2, (int)iStart);
+
+        KernelRange global_range(nRows);
+
+        context.run(global_range, kernel, args, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    return status;
+}
+
+template <typename algorithmFPType, decision_forest::regression::training::Method method>
+services::Status RegressionTrainBatchKernelOneAPI<algorithmFPType, method>::updateMDIVarImportance(const UniversalBuffer & nodeList,
+                                                                                                   const UniversalBuffer & nodeImpDecreaseList,
+                                                                                                   size_t nNodes, UniversalBuffer & varImp,
+                                                                                                   size_t nFeatures)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.updateMDIVarImportance);
+
+    services::Status status;
+
+    auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
+
+    auto & kernel = kernelUpdateMDIVarImportance;
+
+    {
+        KernelArguments args(4);
+        args.set(0, nodeList, AccessModeIds::read);
+        args.set(1, nodeImpDecreaseList, AccessModeIds::read);
+        args.set(2, (int)nNodes);
+        args.set(3, varImp, AccessModeIds::write);
+
+        int localSize = _preferableGroupSize;
+        //calculating local size in way to have all subgroups for node in one group to use local buffer
+        while (localSize > nNodes && localSize > _preferableSubGroup)
+        {
+            localSize >>= 1;
+        }
+
+        KernelRange local_range(localSize, 1);
+        KernelRange global_range(localSize, nFeatures);
+
+        KernelNDRange range(2);
+        range.local(local_range, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+        range.global(global_range, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        context.run(range, kernel, args, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    return status;
+}
+
+template <CpuType cpu>
```

Comment:
Which CpuTypes this code will generate?

Reply:
in all places i take it with sse2. These will gone away when GPU predict layer and GPU generators is added.